### PR TITLE
Update Terraform provider to 3.x

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

This PR updates Terraform provider version to `v3.71.0` which supports the `aws_cloudfront_response_headers_policy` resource. Some existing resources have API changes with this version. I have also tried the minimum required version of `v3.64.0` but we still have some API changes in existing resources, so I went with the latest released version of the Terraform provider in this PR.

Note: I had to exclude `aws_elasticache_user` due to https://github.com/crossplane/terrajet/issues/100, for which a fix is under way.

Note: I just regenerated the Terraformed (managed) resources. No reconfiguration or no adjustments on the examples have been performed.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Ran `make generate` & `make build`.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: cmd/provider/main.go [hunk 1]
```diff
@@ -22,12 +22,12 @@ import (
...
 	"github.com/crossplane/crossplane-runtime/pkg/logging"
 	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
-	tf "github.com/terraform-providers/terraform-provider-aws/aws"
+	"github.com/crossplane/terrajet/pkg/terraform"
 	"gopkg.in/alecthomas/kingpin.v2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
-	"github.com/crossplane/terrajet/pkg/terraform"
+	tf "github.com/hashicorp/terraform-provider-aws/xpprovider"
```
Comment (muvaf) [inline cmd/provider/main.go:30]: Probably unintentional.

File: package/crds/elasticache.aws.jet.crossplane.io_usergroups.yaml [hunk 2]
```diff
@@ -1,212 +0,0 @@
-apiVersion: apiextensions.k8s.io/v1
```
Comment (muvaf) [inline package/crds/elasticache.aws.jet.crossplane.io_usergroups.yaml:1]: Did you intend to remove this as well?

**Review by muvaf — APPROVED** [review]
LGTM, I only have a few small issues. Scanned and didn't see any breaking changes.

Please also make sure that `make reviewable` passes with the full set of resources.

**Comment by muvaf** [comment]
> Note: I had to exclude aws_elasticache_user due to crossplane/terrajet#100, for which a fix is under way.

That issue is fixed now, can we re-include it?

**Comment by maltejk** [comment]
Is this still active?

**Comment by muvaf** [comment]
With https://github.com/crossplane-contrib/provider-jet-aws/pull/169 , we don't need to apply this hack to upgrade the provider anymore. @ulucinar you might want to do this in a clean PR rebased from `main`.

**Comment by haarchri** [comment]
any progress here ? did we need only to bump https://github.com/crossplane-contrib/provider-jet-aws/blob/main/Makefile#L9 ? @ulucinar 

```
make generate                        
23:22:32 [ .. ] generating provider schema for hashicorp/aws 3.62.0
make[1]: *** [config/schema.json] Error 1
make: *** [generate] Error 2
```

we need **LBTargetGroup targetType alb** which was added in v3.62.0

```
---
apiVersion: elbv2.aws.jet.crossplane.io/v1alpha2
kind: LBTargetGroup
metadata:
  name: sample-alb-targetgroup
spec:
  forProvider:
    region: us-west-1
    name: sample-alb-targetgroup
    targetType: alb
    port: 80
    protocol: "TCP"
    vpcIdRef: 
      name: sample-vpc
  providerConfigRef:
    name: default
```
we can add the example afterwards in https://github.com/crossplane-contrib/provider-jet-aws/pull/211

**Comment by ulucinar** [comment]
Hi @haarchri,
Now that we generate the Crossplane provider from JSON schema instead of the native provider source, I think we can close this PR.

**Comment by haarchri** [comment]
Yes but simpler Change im makefile to 3.62.0 for example is not working as you can See in my latest comment @ulucinar

**Pull request closed without merging**